configure.ac

```diff
@@ -2387,8 +2383,16 @@ AS_IF([test "x$enable_zph_qos" = "xyes"],[
     [Enable support for QOS netfilter mark preservation])
 ])
 
-AC_CHECK_LIB(regex, regexec, [REGEXLIB="-lregex"],[REGEXLIB=''])
-AC_SUBST(REGEXLIB)
+dnl Look for libregex with regcomp() API
+SQUID_AUTO_LIB(regex,[GNU Regex],[LIBREGEX])
+SQUID_CHECK_LIB_WORKS(regex,[
+  PKG_CHECK_MODULES([LIBREGEX],[libregex],[:],[:])
```

Comment:
Hm.. what if the local install doesn't have a pkg-config file? For instance, my install of gnurx doesn't have one

Reply:
AFAICT, without pkg-config file, this PKG_CHECK_MODULES() call does nothing (but reporting) because its action-if-not-found is set to "do nothing" (i.e. `:`). The code below will add spaces to CPPFLAGS and LIBS (unless those LIBREGEX_FOO variables were set externally, I guess).

Reply:
Absent a relevant `.pc` file **or** when environment variables are preset this line has no effect.

This line exists solely for auto-detecting the cases when a libregex does provide a `.pc` file.